src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+    )
+
+    orbitals = SubSection(sub_section=APWBaseOrbital.m_def, repeats=True)
+
+    def _determine_apw(self) -> dict[str, int]:
+        """
+        Produce a count of the APW components in the l-channel.
+        Invokes `normalize` on `orbitals`.
+        """
+        for orb in self.orbitals:
+            orb.normalize(None, None)
+
+        type_count = {'apw': 0, 'lapw': 0, 'slapw': 0, 'lo': 0, 'other': 0}
+        for orb in self.orbitals:
+            if isinstance(orb, APWOrbital) and orb.type.lower() in type_count.keys():
+                type_count[orb.type] += 1
+            elif isinstance(orb, APWLocalOrbital):
+                type_count['lo'] += 1
+            else:
+                type_count['other'] += 1  # other de facto operates as a catch-all
+        return type_count
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_normalized = False
+
+    @check_normalized
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super(BasisSet).normalize(archive, logger)
+        self.n_orbitals = len(self.orbitals)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_normalized = False
+
+
+class MuffinTinRegion(BasisSet, Mesh):
+    """
+    Muffin-tin region around atoms, containing the augmented part of the APW basis set.
+    The latter is structured by l-channel. Each channel contains a base (S)(L)APW definition,
+    which may be extended via local orbitals.
+    """
+
+    # there are 2 main ways of structuring the APW basis set
+    # either as APW and lo in the MT region
+    # or by l-channel in the MT region
+
+    radius = Quantity(
+        type=np.float64,
+        shape=[],
+        unit='meter',
+        description="""
+        The radius descriptor of the `MuffinTin` is spherical shape.
+        """,
+    )
+
+    l_max = Quantity(
+        type=np.int32,
+        description="""
+        Maximum angular momentum quantum number that is sampled.
+        Starts at 0.
+        """,
+    )
+
+    l_channels = SubSection(sub_section=APWLChannel.m_def, repeats=True)
+
+    def _determine_apw(self) -> dict[str, int]:
+        """
+        Aggregate the APW component count in the muffin-tin region.
+        Invokes `normalize` on `l_channels`.
+        """
+        for l_channel in self.l_channels:
+            l_channel.normalize(None, None)
+
+        type_count: dict[str, int]
+        if len(self.l_channels) > 0:
+            l_channel = self.l_channels[0]
+            # dynamically determine `type_count` structure
+            type_count = l_channel._determine_apw(l_channel.orbitals)
+            for channel in self.l_channels[1:]:
+                type_count.update(channel._determine_apw())
+        return type_count
+
+    @check_normalized
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+
+class BasisSetContainer(NumericalSettings):
```

Comment:
I am confused: `BasisSet` is also a `NumericalSettings`. So this might get confusing for people trying to understand what info to store under `model_method.numerical_settings`. 

Before, we decided that only the container inherits from it so that people can append it to the `ModelMethod` archive. What has changed?

Reply:
This was a mistake. I had to restructure a couple of times. Has been corrected.